backend/data_controller.py

```diff
@@ -0,0 +1,46 @@
+from google.protobuf.json_format import MessageToDict
+from grpc.messages import data_pb2
+import config
+from flask_restful import Resource
+from flask import request, Flask
+import csv
+from google.protobuf.timestamp_pb2 import Timestamp
+from datetime import datetime
+
+saved_data = {}
+app = Flask(__name__)
+
+
+class Data(Resource):
+    def get(self, filename):
+        page = int(request.args.get('page')) if request.args.get('page') else 0
```

Comment:
Can this be non-numeric input? Maybe also check str.isdigit() or catch ValueError exception on failed conversion.